newIDE/app/src/InstancesEditor/index.js

```diff
@@ -848,13 +861,19 @@ export default class InstancesEditor extends Component<Props> {
       const idealZoom = this.viewPosition.fitToRectangle(contentAABB);
       this.setZoomFactor(idealZoom);
     }
+    if (this.props.onViewPositionChanged) {
+      this.props.onViewPositionChanged(this.viewPosition);
+    }
```

Comment:
maybe this logic you've done 3 times could be in the `setZoomFactor()` function

Reply:
(we need to check in this case the other call sites of setZoomFactor to be sure, if there are any, that it won't impact them)

Reply:
Well the need to call onViewPositionChanged is not related to zoom factor, but the view position, so I'll look into calling it in position change functions